src/v/storage/tests/log_segment_appender_test.cc

```diff
@@ -217,3 +217,40 @@ SEASTAR_THREAD_TEST_CASE(test_can_append_little_data) {
     BOOST_REQUIRE_EQUAL(appender.file_byte_offset(), data.size());
     appender.close().get();
 }
+
+SEASTAR_THREAD_TEST_CASE(test_fallocate_size) {
+    for (const size_t fallocate_size : {1ul, 16_KiB, 32_MiB}) {
```

Comment:
what validation is applied to this new parameter? can I use `0` here or non-power-of-two values? I think it is a good idea to have validation (if we don't already have that--perhaps I missed it), and make sure the tests here have a few edge cases in the space of allowed values.

Reply:
We can use any value > 0. But in segment_appender fallocation_offset will be aligned to 4096.
So I think that here will be better to have 4096 divisor in order to have page aligned fallocations
I'll add validator

Reply:
You may find the existing `storage::internal::chunk_cache::validate_chunk_size` useful as an example (currently used for the `append_chunk_size` setting).

It would be good to have some upper and lower bounds on this to prevent craziness if the user types it wrong: minimum should be the append_chunk_size (if we can't get at the live value from the validator then use the default 16kib) and max should be something arbitrary, maybe the default segment size (1gib).  Arbitrary limits are kind of annoying, but it's better than enabling the user to instantly fill their disk by typing too many zeros in a config option.